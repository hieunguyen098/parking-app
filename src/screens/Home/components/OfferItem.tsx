import { Image, StyleSheet, Text, View } from 'react-native';
import React from 'react';

const OfferItem = () => {
    return (
        <View style={styles.container}>
            <Image
                style={styles.image}
                source={{ uri: 'http://img.giftpop.vn/brand/GIFTPOP/MP1807250002_BASIC_origin.jpg' }}
            />
            <Text style={styles.text}>Ưu đãi 30.000 hấp dẫn!! Nhanh tay nhận voucher</Text>
        </View>
    );
};

export default OfferItem;

const styles = StyleSheet.create({
    container: { marginBottom: 16 },
    image: {
        width: '100%',
        height: 200,
        marginVertical: 8,
        borderRadius: 8,
    },
    text: {
        fontSize: 20,
        fontWeight: '400',
        color: '#404040',
    },
});
