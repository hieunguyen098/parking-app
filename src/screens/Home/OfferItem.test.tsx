import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';

import OfferItem from './components/OfferItem';

test('test offer item', () => {
    const component = (
        <NavigationContainer>
            <OfferItem
                item={{
                    image: 'https://hostingviet.vn/data/tinymce/tin%20tuc%202019/voucher-la-gi.jpg',
                    title: 'Nhận ngay vouher giảm giá lên tới 50%',
                }}
            />
        </NavigationContainer>
    );

    const tree = render(component);
    expect(tree).toMatchSnapshot();
});
