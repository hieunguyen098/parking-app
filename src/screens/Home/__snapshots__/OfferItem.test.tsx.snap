// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test offer item 1`] = `
<View
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://hostingviet.vn/data/tinymce/tin%20tuc%202019/voucher-la-gi.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 8,
        "height": 200,
        "marginVertical": 8,
        "width": "100%",
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#404040",
        "fontSize": 20,
        "fontWeight": "400",
      }
    }
  >
    Nhận ngay vouher giảm giá lên tới 50%
  </Text>
</View>
`;
